### PR TITLE
Warnings fixes

### PR DESCRIPTION
**Pull request by tigerw**
* Fixed some warnings and a logic error in cMonster

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -326,4 +326,4 @@ endif ()
 if (WIN32)
 	target_link_libraries(${EXECUTABLE} expat tolualib ws2_32.lib Psapi.lib)
 endif()
-target_link_libraries(${EXECUTABLE} luaexpat jsoncpp mbedtls zlib sqlite lua SQLiteCpp event_core event_extra)
+target_link_libraries(${EXECUTABLE} luaexpat jsoncpp mbedtls zlib sqlite lua SQLiteCpp event_core event_extra)
```
Comment (madmaxoft) [inline src/CMakeLists.txt]: Missing an endline.

**Comment by madmaxoft** [comment]
Fix the missing endlines and squash the non-working commits at the end into a single one.

**Comment by tigerw** [comment]
The commits all target different things, should they be under a unified commit?

**Comment by NiLSPACE** [comment]
Appveyor failed.

**Comment by tigerw** [comment]
I'm not quite sure how to fix the AppVeyor error. The problem is that JsonCPP generates its library at lib/jsoncpp/src/lib_json/(Debug/Release) and for me, the Visual Studio Linker can't find it without a `link_directories`. This I have done, and confirmed working locally, yet the environment at AppVeyor still seems to be incorrectly configured.

Could it be a cmake version issue?

**Comment by madmaxoft** [comment]
Make a separate PR for each of those unrelated changes. hat way we can merge those that work rightaway, and fiddle with the rest later.

To do so, create a new branch off of the master branch and cherry-pick the specific commits onto it. Amend the commits if they don't compile.

**Comment by tigerw** [comment]
Tada!

**Comment by SafwatHalaby** [comment]
Since this fixes crashes, it should be merged with "testing" and then propagated back to master.

**Comment by madmaxoft** [comment]
I don't quite like the idea of having to decide between two branches, where to merge. I'd say make PRs against `master`, then cherry-pick whatever is needed to `testing`.

**Comment by SafwatHalaby** [comment]
I agree, sounds better.

**Comment by madmaxoft** [comment]
Next time, please don't merge master; rather, rebase (or cherry-pick) onto master. The history is then much cleaner, no more "merged master" commits.

**Comment by madmaxoft** [comment]
Merge if it compiles.

**Comment by SafwatHalaby** [comment]
I didn't merge master (Or I think I didn't, did I?)

**Comment by tigerw** [comment]
> @tigerw Next time, please don't merge master; rather, rebase (or cherry-pick) onto master. The history is then much cleaner, no more "merged master" commits.

Okay.

**Pull request merged**